packages/runtime/container-runtime/src/opLifecycle/outbox.ts

```diff
@@ -32,40 +38,74 @@ export interface IOutboxParameters {
 }
 
 export class Outbox {
+	private readonly mc: MonitoringContext;
 	private readonly attachFlowBatch: BatchManager;
 	private readonly mainBatch: BatchManager;
-	private readonly defaultAttachFlowSoftLimitInBytes = 64 * 1024;
+	private readonly defaultAttachFlowSoftLimitInBytes = 320 * 1024;
+
+	/**
+	 * Track the number of ops which were detected to have a mismatched
+	 * reference sequence number, in order to self-throttle the telemetry events.
+	 *
+	 * This should be removed as part of ADO:2322
+	 */
+	private readonly maxMismatchedOpsToReport = 3;
+	private mismatchedOpsReported = 0;
 
 	constructor(private readonly params: IOutboxParameters) {
+		this.mc = loggerToMonitoringContext(ChildLogger.create(params.logger, "Outbox"));
 		const isCompressionEnabled =
 			this.params.config.compressionOptions.minimumBatchSizeInBytes !==
 			Number.POSITIVE_INFINITY;
 		// We need to allow infinite size batches if we enable compression
 		const hardLimit = isCompressionEnabled ? Infinity : this.params.config.maxBatchSizeInBytes;
 		const softLimit = isCompressionEnabled ? Infinity : this.defaultAttachFlowSoftLimitInBytes;
 
-		this.attachFlowBatch = new BatchManager(
-			{
-				hardLimit,
-				softLimit,
-				enableOpReentryCheck: params.config.enableOpReentryCheck,
-			},
-			params.logger,
-		);
-		this.mainBatch = new BatchManager(
-			{
-				hardLimit,
-				enableOpReentryCheck: params.config.enableOpReentryCheck,
-			},
-			params.logger,
-		);
+		this.attachFlowBatch = new BatchManager({ hardLimit, softLimit });
+		this.mainBatch = new BatchManager({ hardLimit });
 	}
 
 	public get isEmpty(): boolean {
 		return this.attachFlowBatch.length === 0 && this.mainBatch.length === 0;
 	}
 
+	/**
+	 * If we detect that the reference sequence number of the incoming message does not match
+	 * what was already in the batch manager, this means that the batch has been interrupted so
+	 * we will flush the accumulated messages to account for that and create a new batch with the new
+	 * message as the first message.
+	 *
+	 * @param batchManager - the batch manager where the message is supposed to be pushed to
+	 * @param message - the incoming message
+	 */
+	private maybeFlushPartialBatch(batchManager: BatchManager, message: BatchMessage) {
+		const batchReference = batchManager.referenceSequenceNumber;
+		if (batchReference === undefined || batchReference === message.referenceSequenceNumber) {
+			// The reference sequence numbers are stable, there is nothing to do
+			return;
+		}
+
+		if (++this.mismatchedOpsReported <= this.maxMismatchedOpsToReport) {
+			this.mc.logger.sendErrorEvent(
+				{
+					eventName: "ReferenceSequenceNumberMismatch",
```

Comment:
We know this is possible right so this should not be an error. Are we logging this to understand how often this happens?

Reply:
Yes. This was moved from the batch manager.

Reply:
I partially agree with you here. I'd say - overall we do not control scheduling of micro-tasks, so these conditions will happen in natural flow of things.
That said, I think that most of these things today happen due to reentrnacy, i.e., sending ops from DDS event handlers (while ops are applied). It would be awesome if we can fully disallow this flow and validate that helps (statistically).

Ideally we do all of that before starting to play with 0ms timer-based flushes....

Reply:
_I think that most of these things today happen due to reentrnacy_

If so, how can we identify if that happens. Right now, this log will not help us distinguish re-entrancy vs real cases where ops are scheduled via Promises.

Reply:
Oh, I see what you mean, Navin. We can, actually. Check out `verifyCanSubmitOps` in ContainerRuntime. If it's due to op reentry from the DDS event handlers, we will know.